core/lib/src/outcome.rs

```diff
@@ -484,6 +484,102 @@ impl<S, E, F> Outcome<S, E, F> {
         }
     }
 
+    /// Maps an `Outcome<S, E, F>` to an `Outcome<T, E, F>` by applying the
+    /// function `f` to the value of type `S` in `self` if `self` is an
+    /// `Outcome::Success`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use rocket::outcome::Outcome;
+    /// # use rocket::outcome::Outcome::*;
+    /// #
+    /// let x: Outcome<i32, &str, usize> = Success(10);
+    ///
+    /// let mapped = x.and_then(|v| {
```

Comment:
Use a match to make this example (and the other two) clearer:

```rust
let mapped = x.and_then(|v| match v {
    10 => Success("10"),
    1 => Forward(20),
    _ => Failure(false)
});
```

Reply:
I guess you mean 
```rust
let mapped = x.and_then(|v| match v {
   10 => Success("10"),
   1 => Forward(false),
   _ => Failure("30")
});
```
The third generic is the type of `Forward` which is now a bool ;)